packages/engine/paima-runtime/src/runtime-loops.ts

```diff
@@ -264,10 +275,29 @@ async function runSync(
     // Iterate through all of the returned chainData and process each one via the state machine's STF
     try {
       for (const chainData of latestChainDataList) {
-        if (
-          !(await processSyncBlockData(gameStateMachine, chainData, pollingPeriod, stopBlockHeight))
-        ) {
-          break;
+        let count = await processSyncBlockData(
+          gameStateMachine,
+          chainData,
+          pollingPeriod,
+          stopBlockHeight
+        );
+        if (count === -1) break;
+
+        let emulated: number | undefined;
+        let block_chain: number = chainData.blockNumber;
+
+        if (count) {
+          if (ENV.EMULATED_BLOCKS) {
+            const [e] = await emulatedSelectLatestPrior.run(
+              {
+                emulated_block_height: chainData.blockNumber,
+              },
+              gameStateMachine.getReadWriteDbConn()
+            );
+            emulated = block_chain;
+            block_chain = e.deployment_chain_block_height;
+          }
+          MQTTPublisher.sendMessage({ block: block_chain, emulated }, MQTTSystemEvents.STF);
```

Comment:
This event looks like it can be triggered multiple times for the same block. For the format of these topics, I think using the `Separated topics` section in https://github.com/PaimaStudios/paima-engine/issues/373 makes more sense